frontend/src/libs/components/icon/icon.tsx

```diff
@@ -1,10 +1,11 @@
-import { type IconName } from '#libs/types/types.js';
+import { type IconColor } from '#libs/enums/enums.js';
+import { type IconName, type ValueOf } from '#libs/types/types.js';
 
 import { iconNameToIcon } from './libs/maps/icon-name-to-icon.map.js';
 
 type Properties = {
   name: IconName;
-  color?: string;
+  color?: ValueOf<typeof IconColor> | undefined;
```

Comment:
```suggestion
  color?: ValueOf<typeof IconColor>;
```

Reply:
Yeah, that's it. According to the `"exactOptionalPropertyTypes": true` rule in `tsconfig` we can't pass undefined property, even if it has `?`. We can only pass mandatory `ValueOf<typeof IconColor>` or not call that attribute at all